### ISSUE TITLE
Relationship: add BASED_ON or IMPLEMENTATION_OF

### ISSUE THREAD

**Issue by chandanbn** [spdx/spdx-spec#100]
I could not find an existing suitable value for Relationship in following use cases:

Use case 1: Software A is written in Perl. Someone translates it to Python calls it B.

```
B is BASED_ON A.
   or 
B is an IMPLEMENTATION_OF A.
```

Use case 2: There is a specification A (for eg., RFC, patent disclosure, a schema definition). Software B is written based on that specification.

```
B is BASED_ON A
  or
B is an IMPLEMENTATION_OF A
```

Use case 3: Software C contains some functionality from software A and some from B along with new functionality.

```
C is BASED_ON A
C is BASED_ON B
```

**Comment by goneall** [comment]
I just looked back through the relationship definitions and I agree with needing to add additional relationships to cover these uses cases.  I like IMPLEMENTATION_OF for use case 2.  I don't see a problem with `BASED_ON` for use case 1 and 3, but I wonder if there may be a more descriptive term (e.g. `EXTENSION_OF`, `TRANSLATION_OF` or `DERIVATIVE_OF`).

**Comment by swinslow** [comment]
For use case 2, I'm not sure this fits within SPDX's scope for a couple of reasons:

1) SPDX's primary purpose (as I understand it!) is to communicate software bill of material information -- e.g. what files and packages comprise a distribution of software, and what licenses, copyright notices, etc. apply to that distribution. Whether or not the software fulfills a certain grouping of functional attributes seems like it's a very different problem set.

2) Relationships are used to express relationships between SPDX elements -- e.g., between two Packages, or a Package and a File, or an SPDX document and a Package, etc. A "specification" isn't an SPDX element. It defines some set of software functionality, which a software package might implement (or, might or might not implement correctly, ...)

For use cases 1 and 3, I could see those potentially being useful within SPDX's scope and goals, but I wonder if it isn't already captured (at least in part) by the ANCESTOR_OF / DESCENDANT_OF Relationship types.

**Comment by kestewart** [comment]
Moving this to 3.0, since no pull requests have been provided for further discussion.

**Comment by swinslow** [comment]
Agreed, this doesn't feel ripe for 2.2 yet, probably needs more discussion whether these relationships should be added.

**Comment by goneall** [comment]
moving to 3.1 as it is not a breaking change

**Comment by bact** [comment]
Will this related to "derived from" proposed here?:

- https://github.com/spdx/spdx-spec/issues/1227#issuecomment-3325896319

**Comment by goneall** [comment]
Discussed on the 24 March 2026:
Based on the discussion for issue #1227 we agreed to close this issue as already covered by existing relationship types.
